### DISCUSSION TITLE
Contextual floating items positioning

### DISCUSSION THREAD

**Discussion by Moshyfawn** [zed-industries/zed#23925] [Feature Requests]
This is probably more in the category of nit-picking, but it might be a good future feature to add to the overall level of polish of the UI.

Currently, there are instances where tooltips are stacked and overlap each other as well as other popovers covering some of the information underneath. I'd like to see some sort of "smart" positioning system for these instances.

In this example, the ideal behaviour would be for the "More" tooltip to be positioned above the message actions row so that it doesn't cover the actions popover menu.

<img width="240" alt="image" src="https://github.com/zed-industries/zed/assets/16290753/16c7868b-7597-4908-83e3-b3edcba7345b">

<img width="206" alt="image" src="https://github.com/zed-industries/zed/assets/16290753/a13da510-cc35-454f-a9b0-d96639395d03">

**Comment by maxdeviant** [comment]
This is definitely a tricky thing to solve! I think this will require more discussion on how to best handle it.
